x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/rest/action/role/RestBulkPutRolesAction.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+package org.elasticsearch.xpack.security.rest.action.role;
+
+import org.elasticsearch.action.support.WriteRequest;
+import org.elasticsearch.client.internal.node.NodeClient;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.license.XPackLicenseState;
+import org.elasticsearch.rest.RestRequest;
+import org.elasticsearch.rest.action.RestToXContentListener;
+import org.elasticsearch.xpack.core.security.action.role.BulkPutRoleRequestBuilder;
+import org.elasticsearch.xpack.core.security.action.role.BulkPutRoleRequestBuilderFactory;
+
+import java.io.IOException;
+import java.util.List;
+
+import static org.elasticsearch.rest.RestRequest.Method.POST;
+
+/**
+ * Rest endpoint to bulk add a Roles to the security index
+ */
+public class RestBulkPutRolesAction extends NativeRoleBaseRestHandler {
+
+    private final BulkPutRoleRequestBuilderFactory builderFactory;
+
+    public RestBulkPutRolesAction(Settings settings, XPackLicenseState licenseState, BulkPutRoleRequestBuilderFactory builderFactory) {
+        super(settings, licenseState);
+        this.builderFactory = builderFactory;
+    }
+
+    @Override
+    public List<Route> routes() {
+        return List.of(Route.builder(POST, "/_security/_bulk/role").build());
```

Comment:
I don't like the URL path here.
Things that start with `_` (like `_bulk` or `_security`) can be used for path elements that otherwise would be captured to variables (for eg `_security` could otherwise be captured as a variable for an index name).
I would simply suggest `POST _security/role`, or `POST _security/role/_bulk_update`. I think I'll ask the team as well.

Reply:
Decided on: `POST /_security/role`